libraries/entities-renderer/src/EntityTreeRenderer.cpp

```diff
@@ -167,6 +167,13 @@ void EntityTreeRenderer::resetEntitiesScriptEngine() {
     auto entityScriptingInterface = DependencyManager::get<EntityScriptingInterface>();
     entityScriptingInterface->setEntitiesScriptEngine(entitiesScriptEngineProvider);
 
+    connect(_entitiesScriptEngine.data(), &ScriptEngine::entityScriptPreloadFinished, [&](const EntityItemID& entityID) {
+        EntityItemPointer entity = getTree()->findEntityByID(entityID);
+        if (entity) {
+            entity->setScriptHasFinishedPreload(true);
+        }
+    });
+
     // Connect mouse events to entity script callbacks
     if (!_mouseAndPreloadSignalHandlersConnected) {
```

Comment:
I've been suspicious of this if check for a while.  are we sure all the other mouse events that are connected below work after a reload?  I think if we just removed the third argument, I would expect them to, but with the third argument being `_entitiesScriptEngine.data()`, I would expect the signal to be disconnected in the same way when the script engine is reset

Reply:
Yes, the mouse events below work after reload - even though `_mouseAndPreloadSignalHandlersConnected == true`  still after reload. (Which is why the code calling `entity->setSciptHasFinishedPreload(true);` had to be moved - to connect the new `_entitiesScriptEngine`'s signal `entityScriptPreloadFinished`, and so set things up such that `enterEntty` and `leaveEntity` can fire.

The following test script demonstrates this...

[enterEntity.zip](https://github.com/kasenvr/project-athena/files/5503762/enterEntity.zip)